cmake/macros/Python.cmake

```diff
@@ -51,7 +51,14 @@ function(add_python_target target)
   target_link_libraries(${target} PKG::PYTHON)
 
   if(BUILD_SHARED_LIBS)
-    set(_outdir "${PROJECT_BINARY_DIR}/${CMAKE_CFG_INTDIR}/${slash_namespace}")
+    if(CMAKE_GENERATOR STREQUAL "Xcode")
+      # This is explained in CompilerFlags.cmake
```

Comment:
Code duplication scares me. I'm wondering if instead `intdir` should be encouraged for global use (e.g. defined globally in CompilerFlags and given an `UPPERCASE_WITH_UNDERSCORES` name) and we just use that everywhere that `CMAKE_CFG_INTDIR` was used previously.

Reply:
I actually did do something like this on my iOS branch to fix an issue with $(EFFECTIVE_PLATFORM_NAME), I must have forgotten to move it over. But the code duplication in this case comes from the fact that make would spam warnings if the output directory was set to `CMAKE_CFG_INTDIR`. Making it a generator expression fixed it in these two cases, but we can't have this generator expression being used everywhere we're using `CMAKE_CFG_INTDIR`. This code duplication doesn't bother me as much since it's only happening two times though, both in order to bypass the same warnings.